src/stores/user-store.js

```diff
@@ -11,30 +11,24 @@ class UserStore {
     return this.collection.find().toArray();
   }
 
-  async findUserByLogin({ provider, emailOrUsername }, { session } = {}) {
-    const lowerCasedEmailOrUsername = emailOrUsername?.toLowerCase() || '';
-    const matches = await this.collection.find({
-      $and: [
-        { provider },
-        { $or: [{ username: emailOrUsername }, { email: lowerCasedEmailOrUsername }] }
-      ]
-    }, {
-      session
-    }).toArray();
-
-    let bestMatch;
-    switch (matches.length) {
-      case 0:
-        bestMatch = null;
-        break;
-      case 1:
-        bestMatch = matches[0];
-        break;
-      default:
-        bestMatch = matches.find(match => match.email === lowerCasedEmailOrUsername);
+  findUsersByEmailOrUsername({ provider, email, username }, { session } = {}) {
+    const queryFilters = [];
+
+    if (email) {
+      queryFilters.push({ email });
+    }
+
+    if (username) {
+      queryFilters.push({ username });
+    }
+
+    if (!queryFilters.length) {
+      return [];
     }
 
-    return bestMatch;
+    return this.collection
+      .find({ $and: [{ provider }, queryFilters.length === 1 ? queryFilters[0] : { $or: queryFilters }] }, { session })
```

Comment:
For the sake of simplification I think here you can simply say:
`.find({ $and: [{ provider },  { $or: queryFilters }] }, { session })`
even for the case of length being one

Reply:
True, that was a bit over-dogmatic. Simplified it, thanks!